sw/host/opentitansession/src/main.rs

```diff
@@ -108,5 +238,39 @@ fn main() -> Result<()> {
         return Ok(());
     }
 
-    unimplemented!("Background daemon not implemented");
+    if opts.child {
+        // This process is a child, which is supposed to stay running as a daemon.
+        match session_child(opts.listen_port, &opts.backend_opts) {
+            Ok(()) => process::exit(0),
+            Err(e) => {
+                // Report any error to parent process though stdout pipe.
+                serde_json::to_writer::<io::Stdout, Result<SessionStartResult, String>>(
+                    io::stdout(),
+                    &Err(format!("{}", e).to_string()),
+                )?;
+                process::exit(1)
+            }
+        }
+    }
+
+    // Locate directory to use for .pid files
```

Comment:
An alternative that does not require parsing a PID and fussing with signals is to have the child monitor the file periodically; if the file ceases to exist (because someone deleted it) the daemon can kill itself. You lose confirmation of daemon destruction but I don't think that's terrifically critical.

Reply:
I agree that you would remove some complexity by avoiding signals, but I do think the benefits are worthwhile.

I imagine testing scripts might start a session, run a sequence of commands that interact with the session, and then stop the session, before going on to the next round/script.  It would be advantageous to guarantee that the session daemon has finished releasing its handles to the USB debugger device, by the time the "end session" command terminates, such that one can be sure that a "start session" very soon after, would be able to successfully make connection to the same debugger.
Also, I find the mio support for signals rather nice. The current daemon implementation is fully event driven, that is, it does not need to repeatedly wake up the CPU to do anything (like checking file timestamps), it only reacts to events as they happen, sitting in a epoll() call without any timeout otherwise.

Reply:
Fair I guess. I detest messing with unix stuff directly but... I'll allow it, as long as it's contained as it is here.

Reply:
Thank you.  I had thought about way of eliminating mio_signal from being explicitly used in the main loop in opentitanlib/src/proxy, by letting the caller pass a mio::Source and callback function, which this file would use for passing a signal handler.  In that way, there would be no mention of Unix signals in opentitanlib, only in this one file.  I might still propose a PR with such a refactoring.  Then should we later decide to add support for Windows, it would probably be by a separate main_win.rs file in this directory, and it would likely directly refer to Windows interprocess communication means of requesting the background process to terminate.